server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java

```diff
@@ -1815,4 +1822,48 @@ private void assertGlobalCheckpointIsStableAndSyncedInAllNodes(String indexName,
             assertBusy(() -> assertThat(nodeIndexShard.getLastSyncedGlobalCheckpoint(), equalTo(maxSeqNo)));
         }
     }
+
+    private void createSnapshotThatCanBeUsedDuringRecovery(String indexName) throws Exception {
+        // Ensure that the safe commit == latest commit
+        assertBusy(() -> {
+            ShardStats stats = client().admin().indices().prepareStats(indexName).clear().get()
+                .asMap().entrySet().stream().filter(e -> e.getKey().shardId().getId() == 0)
+                .map(Map.Entry::getValue).findFirst().orElse(null);
+            assertThat(stats, is(notNullValue()));
+            assertThat(stats.getSeqNoStats(), is(notNullValue()));
+
+            assertThat(Strings.toString(stats.getSeqNoStats()),
+                stats.getSeqNoStats().getMaxSeqNo(), equalTo(stats.getSeqNoStats().getGlobalCheckpoint()));
+        }, 60, TimeUnit.SECONDS);
+
+        // Force merge to make sure that the resulting snapshot would contain the same index files as the safe commit
+        ForceMergeResponse forceMergeResponse = client().admin().indices().prepareForceMerge(indexName).setFlush(randomBoolean()).get();
```

Comment:
My goal here was to ensure that we recover from the snapshot, so that's why I'm doing a force merge. Is it possible that we end up with different segment files even after a force merge?

Reply:
I was overinterpreting the location of it here. I think it is good with randomBoolean though, otherwise snapshotting will do the necessary flush for us.